src/main/java/me/zeroeightsix/kami/module/modules/combat/VisualRange.kt

```diff
@@ -59,15 +62,15 @@ internal object VisualRange : Module(
     }
 
     private fun onEnter(player: EntityPlayer) {
-        sendNotification("${getColor(player) format player.name} spotted!")
-        if (logToFile.value) WaypointManager.add(player.flooredPosition, "${player.name} spotted!")
+        sendNotification(String.format(onEnterMessage.value, getColor(player).format(player.name)))
+        if (logToFile.value) WaypointManager.add(player.flooredPosition, String.format(onEnterMessage.value, player.name))
         if (uwuAura.value) sendServerMessage("/w ${player.name} hi uwu")
     }
 
     private fun onLeave(player: EntityPlayer) {
         if (leaving.value) {
-            sendNotification("${getColor(player) format player.name} left!")
-            if (logToFile.value) WaypointManager.add(player.flooredPosition, "${player.name} left!")
+            sendNotification(String.format(onEnterMessage.value, getColor(player).format(player.name)))
+            if (logToFile.value) WaypointManager.add(player.flooredPosition, String.format(onLeaveMessage.value, player.name))
```

Comment:
Given how many times you're using format, maybe have a `private fun StringSetting.replacedName() = this.value.replace("%s", player.name)` method and do `onLeaveMessage.replacedName()`